src/Pester.Runtime.ps1

```diff
@@ -58,6 +58,59 @@ $flags = [System.Reflection.BindingFlags]'Instance,NonPublic'
 $script:SessionStateInternalProperty = [System.Management.Automation.SessionState].GetProperty('Internal', $flags)
 $script:ScriptBlockSessionStateInternalProperty = [System.Management.Automation.ScriptBlock].GetProperty('SessionStateInternal', $flags)
 $script:ScriptBlockSessionStateProperty = [System.Management.Automation.ScriptBlock].GetProperty("SessionState", $flags)
+# function for extracting StacktraceSystemLanguage
+function Get-StackTraceLanguageFallBack {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$StackTrace
+    )
+    $StacktraceLanguage = [PSCustomObject]@{
+        At   = "at"
+        Line = "line"
+    }
+    $regex = "(?<At>.*)\s(?<ScriptBlockOrFunction>\<\w+\>),\s(?<FileName>\<.+\>)\s*:\s(?<Line>\w+)\s(?<LineNumber>\w+)\z"
+    if ($StackTrace -match $regex) {
+        $StacktraceLanguage.At = $Matches["At"]
+        $StacktraceLanguage.Line = $Matches["Line"]
+    }
+    Return $StacktraceLanguage
+}
+
+function Get-StackTraceLanguage {
+    #Full fallback scenario to the solution of PR #2276 in case of error
+    try {
+        if ($PSVersionTable.PSVersion.Major -ge 5) {
+            if ($PSVersionTable.PSVersion.Major -gt 5) {
+                $StackTraceRessourceName = 'System.Management.Automation.resources.DebuggerStrings'
+            }
+            else {
+                $StackTraceRessourceName = 'DebuggerStrings'
+            }
+            $r = [System.Resources.ResourceManager]::new($StackTraceRessourceName, [System.Reflection.Assembly]::GetAssembly([System.Management.Automation.Breakpoint]))
+            $result = $r.GetString('StackTraceFormat') -match '^(?<At>.*) \{0\}, \{1\}: (?<Line>.*) \{2\}$'
+            if (!($result)) {
+                throw 'StackTraceFormat in a unknown format'
+            }
+            else {
+                $StacktraceLanguage = [PSCustomObject]@{
+                    At   = $Matches["At"]
+                    Line = $Matches["Line"]
+                }
+            }
+        }
+        else {
+            #Prior to version 5 there are no ressources in the Assembly available (System.Management.Automation)
+            throw 'Fallback for PSVersion 3/4'
```

Comment:
Using exceptions for control flow is not ideal, this will always throw for powershell 4, if you just reduce the scope of the `try` to be in the positive leg of the if it should make the code no more complicated, and avoid this issue.

Reply:
Second this. Please move fallback exception to be part of `Get-StackTraceLanguageFallBack`. That way we can remove the large try/catch here and just call the fallback if no result or PS <5.

Reply:
Second this. Please move fallback exception to be part of `Get-StackTraceLanguageFallBack`. That way we can remove the large try/catch here and just call the fallback if no result or PS <5.